common/oauthTokenManager.go

```diff
@@ -189,6 +190,7 @@ func (uotm *UserOAuthTokenManager) MSILogin(identityInfo IdentityInfo, persist b
 	oAuthTokenInfo := &OAuthTokenInfo{
 		Identity:     true,
 		IdentityInfo: identityInfo,
+		Persist:      persist,
 	}
 
 	return uotm.validateAndPersistLogin(oAuthTokenInfo, persist)
```

Comment:
- 'persist' is part of oAuthTokenInfo already. 'validateAndPersistLogin' is taking both inputs which is redundant and can be removed.
- 'validateAndPersistLogin' is persisting in credCache. We discussed to always use AzIdentity cache first and if not available to use credCache. Do not see those changes in these flow.

Reply:
Only in device login we want to use AzIdentity cache and no other login like SPN/MSI/SecretaBased login etc.

Reply:
we have removed the redundant code in validateAndPersistLogin, and are now using the persist information in oAuthtokenInfo.
azidentity does not support Authentication Record in SPN/MSI/Secret based login , hence we have not used that flow here.